src/mon/MDSMonitor.cc

```diff
@@ -1489,11 +1520,17 @@ int MDSMonitor::filesystem_command(
     std::string role_str;
     cmd_getval(cmdmap, "role", role_str);
     mds_role_t role;
-    int r = fsmap.parse_role(role_str, &role, ss);
+    const auto& fs_names = op->get_session()->get_allowed_fs_names();
```

Comment:
```suggestion
    const auto fs_names = op->get_session()->get_allowed_fs_names();
```

`get_allowed_fs_names` returns an rvalue. What you're doing is legal but weird.

Reply:
True, fixed.